frontends/smvparser.l

```diff
@@ -0,0 +1,131 @@
+floor {return cosa::smvparser::make_tok_floor();}
+extend {return cosa::smvparser::make_extend();}
+resize {return cosa::smvparser::make_resize();}
+typeof {return cosa::smvparser::make_tok_typeof();}
+
+unsigned   {return cosa::smvparser::make_tok_unsigned();}
+signed   {return cosa::smvparser::make_tok_signed();}
+boolean   {return cosa::smvparser::make_bool_type();}
+real  {return cosa::smvparser::make_real_type();}
+integer   {return cosa::smvparser::make_integer_type();}
+ck   {return cosa::smvparser::make_clock_type();}
+word   {return cosa::smvparser::make_tok_word();}
+set {return cosa::smvparser::make_tok_set();}
+"in" {return cosa::smvparser::make_OP_in();}
+time {return cosa::smvparser::make_time_type();}
+
+"->"  {return cosa::smvparser::make_OP_IMPLY();}
+"<->"  {return cosa::smvparser::make_OP_BI();}
+"!"  {return cosa::smvparser::make_OP_NOT();}
+"::"  {return cosa::smvparser::make_OP_CON();}
+"-"  {return cosa::smvparser::make_OP_MINUS();}
+"+"  {return cosa::smvparser::make_OP_PLUS();}
+"*"  {return cosa::smvparser::make_OP_MUL();}
+"/"  {return cosa::smvparser::make_OP_DIV(); }
+"mod"  {return cosa::smvparser::make_OP_MOD();}
+">>"  {return cosa::smvparser::make_OP_SHIFTR();}
+"<<"  {return cosa::smvparser::make_OP_SHIFTL();}
+"union"  {return cosa::smvparser::make_UNION();}
+"="  {return cosa::smvparser::make_OP_EQ();}
+"!="  {return cosa::smvparser::make_OP_NEQ();}
+">"  {return cosa::smvparser::make_OP_GT();}
+"<"  {return cosa::smvparser::make_OP_LT();}
+"<="  {return cosa::smvparser::make_OP_LTE();}
+">="  {return cosa::smvparser::make_OP_GTE();}
+"&"  {return cosa::smvparser::make_OP_AND();}
+"|"  {return cosa::smvparser::make_OP_OR();}
+"xor"  {return cosa::smvparser::make_OP_XOR();}
+"xnor"  {return cosa::smvparser::make_OP_XNOR();}
+
+".." {return cosa::smvparser::make_TO();}
+"(" {return cosa::smvparser::make_LPARE();}
+")" {return cosa::smvparser::make_RPARE();}
+"[" {return cosa::smvparser::make_LBRA();}
+"]" {return cosa::smvparser::make_RBRA();}
+"{" {return cosa::smvparser::make_LBRACE();}
+"}" {return cosa::smvparser::make_RBRACE();}
+";" {return cosa::smvparser::make_SEMICOLON();}
+":" {return cosa::smvparser::make_COLON();}
+"," {return cosa::smvparser::make_COMMA();}
+":=" {return cosa::smvparser::make_ASSIGNSYM();}
+"--".*  {}
+"?"  {return cosa::smvparser::make_IF_ELSE();}
+"_" {return cosa::smvparser::make_UNDER();}
+
+0(b|d|o|h)/[0-9]+_[0-9]+ {BEGIN(WORD_INIT); return cosa::smvparser::make_word_index1(yytext);}
+0(u|s)(b|d|o|h)/[0-9]+_[0-9]+ {BEGIN(WORD_INIT);  return cosa::smvparser::make_word_index2(yytext);}
+<WORD_INIT>[0-9]+/_[0-9]+ {BEGIN(WORD_BASE); return cosa::smvparser::make_integer_val(atoi(yytext));}
+<WORD_BASE>_/[0-9]+ {BEGIN(WORD_VALUE); return cosa::smvparser::make_UNDER();}
+<WORD_VALUE>[0-9]+ {BEGIN(INITIAL); return cosa::smvparser::make_integer_val(atoi(yytext));}
+
+[0-9]+           {return cosa::smvparser::make_integer_val(atoi(yytext));}
```

Comment:
support for large integers?

Reply:
Hi @yangy96, sorry for not weighing in on this earlier. Thanks @ahmed-irfan for noticing this. If the integer value cannot be represented as a machine int, this will break. Instead, you can just pass the string representation to the `SmtSolver::make_term` function (this will handle larger integers).